hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestHoodieBackedMetadata.java

```diff
@@ -3002,6 +3004,99 @@ public void testOutOfOrderCommits() throws Exception {
     validateMetadata(client);
   }
 
+  @Test
+  public void testDeleteWithRecordIndex() throws Exception {
+    init(HoodieTableType.COPY_ON_WRITE, true);
+    HoodieSparkEngineContext engineContext = new HoodieSparkEngineContext(jsc);
+    HoodieWriteConfig writeConfig = getWriteConfigBuilder(true, true, false)
+        .withMetadataConfig(HoodieMetadataConfig.newBuilder().withEnableRecordIndex(true).withMaxNumDeltaCommitsBeforeCompaction(1).build())
+        // In memory index is required for the writestatus to track the written records
+        .withIndexConfig(HoodieIndexConfig.newBuilder().withIndexType(HoodieIndex.IndexType.INMEMORY).build())
```

Comment:
These 2 options to enable the RLI looks weird, for Spark engine the default index type is `SIMPLE`, that means in order to enable RLI, people always needs to configure these two, shouldn't RLI be a new index type then?